src/core/config/schemas.ts

```diff
@@ -281,7 +285,10 @@ export const LLMConfigSchema = z
         }
     });
 
-export type LLMConfig = z.infer<typeof LLMConfigSchema>;
+// Input type for user-facing API (pre-parsing)
+export type LLMConfig = z.input<typeof LLMConfigSchema>;
+// Validated type for internal use (post-parsing)
+export type ValidatedLLMConfig = z.infer<typeof LLMConfigSchema>;
```

Comment:
@coderabbitai i am considering renaming the convention

maybe we could use InputLLMConfig and RuntimeLLMConfig to distinguish?

what do you think? how does this compare to the existing convention

Reply:
ok i'm still not fully sure about this because Runtime is a bit vague.

maybe we can have a middle ground of InputLLMConfig and ValidatedLLMConfig and document the differences

Reply:
if we don't use the union types, will end-users see type errors in their IDE or typescript config if they use an object with ValidatedLLMConfig when the input accepts InputLLMConfig?